File: loex_Golang/pkg/client/orderwebsocketclient/subscribeorderwebsocketv2client.go
package orderwebsocketclient

import (
	"fmt"
	"loex_golang/logging/applogger"
	"loex_golang/pkg/client/websocketclientbase"
)

// Responsible to handle order subscription from WebSocket
// This need authentication version 2
type SubscribeOrderWebSocketV2Client struct {
	websocketclientbase.WebSocketV2ClientBase
}

// Initializer
func (p *SubscribeOrderWebSocketV2Client) Init(accessKey string, secretKey string, host string) *SubscribeOrderWebSocketV2Client {
	p.WebSocketV2ClientBase.Init(accessKey, secretKey, host)
	return p
}

// Set callback handler
func (p *SubscribeOrderWebSocketV2Client) SetHandler(
	authHandler websocketclientbase.AuthenticationV2ResponseHandler,
	responseHandler websocketclientbase.ResponseHandler) {
	p.WebSocketV2ClientBase.SetHandler(authHandler, p.handleMessage, responseHandler)
}

// Subscribe all order updates of the current account
func (p *SubscribeOrderWebSocketV2Client) Subscribe(symbol string, clientId string) {
	channel := fmt.Sprintf("orders_%s", symbol)
	sub := fmt.Sprintf("{\"action\":\"sub\", \"ch\":\"%s\"}", channel)

	p.Send(sub)

	applogger.Info("WebSocket subscribed, channel=%s, clientId=%s", channel, clientId)
}

// Unsubscribe order updates
func (p *SubscribeOrderWebSocketV2Client) UnSubscribe(symbol string, clientId string) {
	channel := fmt.Sprintf("orders_%s", symbol)
	unsub := fmt.Sprintf("{\"action\":\"unsub\", \"ch\":\"%s\"}", channel)

	p.Send(unsub)

	applogger.Info("WebSocket unsubscribed, channel=%s, clientId=%s", channel, clientId)
}

func (p *SubscribeOrderWebSocketV2Client) handleMessage(msg string) (interface{}, error) {
	return msg, nil
}

File: loex_CSharp/Loex.SDK.Core/Client/OrderWebSocketClient/SubscribeOrderWebSocketV2Client.cs
using Loex.SDK.Core.Client.WebSocketClientBase;
using Loex.SDK.Core.Log;

namespace Loex.SDK.Core.Client
{
    /// <summary>
    /// Responsible to handle order update from WebSocket
    /// This need authentication version 2
    /// </summary>
    public class SubscribeOrderWebSocketV2Client : WebSocketV2ClientBase
    {
        /// <summary>
        /// Constructor
        /// </summary>
        /// <param name="accessKey">API Access Key</param>
        /// <param name="secretKey">API Secret Key</param>
        /// <param name="host">API Host</param>
        public SubscribeOrderWebSocketV2Client(string accessKey, string secretKey, string host = DEFAULT_HOST)
            :base(accessKey, secretKey, host)
        {
        }

        /// <summary>
        /// Subscribe order update for these events: creation, trade, cancellation.
        /// </summary>
        /// <param name="symbol">Trading symbol (wildcard * is allowed)</param>
        /// <param name="clientId"></param>
        public void Subscribe(string symbol, string clientId = "")
        {
            string topic = $"orders_{symbol}";

            _WebSocket.Send($"{{\"action\":\"sub\", \"ch\":\"{topic}\" }}");

            _logger.Log(LogLevel.Info, $"WebSocket subscribed, topic={topic}");
        }

        /// <summary>
        /// Unsubscribe order update
        /// </summary>
        /// <param name="symbol">Trading symbol (wildcard * is allowed)</param>
        /// <param name="clientId">Client id</param>
        public void UnSubscribe(string symbol, string clientId = "")
        {
            string topic = $"orders_{symbol}";

            _WebSocket.Send($"{{\"action\":\"unsub\", \"ch\":\"{topic}\" }}");

            _logger.Log(LogLevel.Info, $"WebSocket unsubscribed, topic={topic}");
        }
    }
}
